container-build/cerc-watcher-erc20/Dockerfile

```diff
@@ -0,0 +1,13 @@
+FROM node:16.17.1-alpine3.16
+
+RUN apk --update --no-cache add git python3 alpine-sdk
+
+WORKDIR /app
+
+COPY . .
+
+RUN echo "Building watcher-ts" && \
+    git checkout v0.2.19 && \
```

Comment:
i get `error: pathspec 'v0.2.19' did not match any file(s) known to git`
even though i see the latest watcher-ts release of `v0.2.19` is over an hour ago. works locally

Reply:
awesome, yep `--pull` did the trick, thanks

Reply:
A special branch should be implemented in `laconic-so setup-repositories`. I think there is already some support for non-main branches there but it's been a while since I tried it.

Reply:
oh i see, `--pull` is only required because this PR isn't merged on to main, makes sense

Reply:
> I think there is already some support for non-main branches there but it's been a while since I tried it.

@dboreham
I see that checking out to branches is currently supported using the `--branches-file` option.
However, when I try checking out to the required release tag, it works if the `HEAD` is pointing to a branch and throws the following error if it's detached (checked out to a tag for eg.):

```
(venv) $ laconic-so --verbose setup-repositories --include cerc-io/watcher-ts --pull --branches-file branches.txt
loading branches from: branches.txt
Branches are: ['cerc-io/watcher-ts v0.2.18']
.
.
  File "/home/prathamesh/deepstack/stack-orchestrator/app/setup_repositories.py", line 116, in process_repo
    present_text = f"already exists active branch: {git.Repo(full_filesystem_repo_path).active_branch}" if is_present \
  File "/home/prathamesh/deepstack/stack-orchestrator/venv/lib/python3.10/site-packages/git/repo/base.py", line 865, in active_branch
    return self.head.reference
  File "/home/prathamesh/deepstack/stack-orchestrator/venv/lib/python3.10/site-packages/git/refs/symbolic.py", line 309, in _get_reference
    raise TypeError("%s is a detached symbolic reference as it points to %r" % (self, sha))
TypeError: HEAD is a detached symbolic reference as it points to '9566dff4664f1f9620d2edd6a3ce78a7c6ca40c3'
```

Reply:
> oh i see, --pull is only required because this PR isn't merged on to main, makes sense

`--pull` is actually required because the repo we're trying to setup has a new remote branch / tag.

Reply:
Hmm...ok. But do we need to support the "already detached" state? Wouldn't the repo have been cloned by stack-orchestrator, hence it wouldn't be detached?

Reply:
> hence it wouldn't be detached

It might be in a detached state when we run `setup-repositories` the next time (re-running the demo for eg.).

Reply:
Ok, this approach is fine until we add proper branch/tag-switching support.